### PR TITLE
bug fixes

### PR DESCRIPTION
**Pull request by alexcos20**
- fix bug on DELETE /compute if no job is found 
 - fix bug on GET /compute if no job is found 
 - fix bug to return [] instead of null is a job is not found (delete/get) 
- added provider signature verification for PUT /compute and GET /compute

### CODE REVIEW

**Comment by alexcos20** [comment]
disabled provider signature check for enpoints:  GET, PUT, DELETE

We need to add another parameter, because in this endpoints, agreementId is optional.  (You can DELETE using a specified jobId or GET using an specified owner)

File: operator_service/routes.py [hunk 1]
```diff
@@ -218,21 +218,19 @@ def stop_compute_job():
...
-        msg, status = process_signature_validation(data.get('providerSignature'), agreement_id)
-        if msg:
-            return jsonify(error=f'`providerSignature` of agreementId is required.'), status
-
         jobs_list = get_sql_jobs(agreement_id, job_id, owner)
-        for ajob in jobs_list:
-            name = ajob
-            logging.info(f'Stopping job : {name}')
-            stop_sql_job(name)
+        if jobs_list is not None:
```
Comment (ssallam) [inline operator_service/routes.py]: It looks cleaner if `get_sql_jobs` always returned a list, then this check is not necessary.

File: operator_service/routes.py [hunk 2]
```diff
@@ -298,25 +296,25 @@ def delete_compute_job():
...
-                name,
-                body,
-                grace_period_seconds=grace_period_seconds,
-                orphan_dependents=orphan_dependents,
-                propagation_policy=propagation_policy
-            )
-            logging.debug(api_response)
-
+        logging.debug(f'Got {jobs_list}')
+        if jobs_list is not None:
```
Comment (ssallam) [inline operator_service/routes.py]: Same as above.

File: operator_service/routes.py [hunk 3]
```diff
@@ -218,21 +218,19 @@ def stop_compute_job():
 
         if not owner or len(owner) < 2:
             owner = None
-
+        #msg, status = process_signature_validation(data.get('providerSignature'), agreement_id)
+        #if msg:
+        #    return jsonify(error=f'`providerSignature` of agreementId is required.'), status
```
Comment (ssallam) [inline operator_service/routes.py]: Should just remove this code block, we can rewrite it later.

**Pull request merged**